docs/services/nifi.md

```diff
@@ -0,0 +1,99 @@
+<!--
+SPDX-FileCopyrightText: 2025 spatterlight
+
+SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+# Apache NiFi
+
+[Apache NiFi](https://nifi.apache.org/) is an open source, easy to use, powerful, and reliable system to process and distribute data.
+
+## Dependencies
+
+This service requires the following other services:
+
+- a [Traefik](traefik.md) reverse-proxy server
+
+## Prerequisites
+
+To deploy Apache NiFi using this role it is necessary that:
+
+1. The [community.general](https://github.com/ansible-collections/community.general) collection be installed. This is needed to support modifying XML configuration files.
+2. The [community.crypto](https://github.com/ansible-collections/community.crypto) collection be installed. This is needed to create the self-signed HTTPS certificate for Apache NiFi.
+3. The `keytool` program be available on the target host. This can be installed via `apt install default-jre` on Debian systems.
```

Comment:
I suppose it's possible to run these tools in containers and avoid these dependencies.

In some roles we run `htpasswd` from some `httpd` container for the same reason.

For SSL certificates, an alternative may be to ask the user to generate the SSL certificate themselves and to provide it via `vars.yml`. Then Ansible would just install it. It makes things a bit harder on the user, but the final installation becomes more reproducible. Maybe it's not ideal, given that it will be harder on users, making them do manual manual work (which is better done by the tool).